src/terminal.c

```diff
@@ -460,21 +430,6 @@ DEFUN ("terminal-list", Fterminal_list, Sterminal_list, 0, 0, 0,
 
   return terminals;
 }
-
-DEFUN ("terminal-name", Fterminal_name, Sterminal_name, 0, 1, 0,
-       doc: /* Return the name of the terminal device TERMINAL.
-It is not guaranteed that the returned value is unique among opened devices.
-
-TERMINAL may be a terminal object, a frame, or nil (meaning the
-selected frame's terminal). */)
-  (Lisp_Object terminal)
-{
-  struct terminal *t = decode_live_terminal (terminal);
-
-  return t->name ? build_string (t->name) : Qnil;
```

Comment:
I can't see the null check of the name in the Rust. Have I missed it, or do we need to add it?

Reply:
This is already checked in `decode_terminal` with `return t && t->name ? t : NULL;`. I used the method `is_live` in rust.